pkg/placement/placement_client.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2022 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package placement
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"path"
+	"time"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	"github.com/pingcap/tidb/store/pdtypes"
+	"github.com/pingcap/tiflow/pkg/config"
+	"github.com/pingcap/tiflow/pkg/httputil"
+	"github.com/pingcap/tiflow/pkg/retry"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+)
+
+const (
+	updateTypeDel = iota
+	updateTypeAdd
+)
+
+const (
+	batchUpdatePrefix = "/pd/api/v1/config/rules/batch"
+	groupQueryPrefix  = "/pd/api/v1/config/rules/group"
+	idQueryPrefix     = "/pd/api/v1/config/rule"
+
+	addMetaJson = `[{
+		"group_id": "pd",
+		"id": "cdcmeta",
+		"index": 20000,
+		"override": true,
+		"start_key": "6d00000000000000f8",
+		"end_key": "6e00000000000000f8",
+		"role": "voter",
+		"count": 3,
+		"action": "add",
+		"delete_by_id_prefix": false
+	}]`
+
+	delCDCGroupJson = `[{
+		"group_id": "pd",
+		"id": "cdcmeta",
+		"action": "del",
+		"delete_by_id_prefix": true
+	}]`
+)
+
+var placmentRuleCheckInterval = 20 * time.Second
+
+type PlacementClient struct {
+	pdClient   pd.Client
+	dialClient *httputil.Client
+
+	ctx    context.Context
+	cancel context.CancelFunc
+	stopCh chan struct{}
+}
+
+func NewPlacementClient(ctx context.Context, pdClient pd.Client) (*PlacementClient, error) {
+	ctx, cancel := context.WithCancel(ctx)
+	_ = cancel
+	conf := config.GetGlobalServerConfig()
+	dialClient, err := httputil.NewClient(conf.Security)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &PlacementClient{
+		pdClient:   pdClient,
+		dialClient: dialClient,
+		ctx:        ctx,
+		cancel:     cancel,
+		stopCh:     make(chan struct{}),
+	}, nil
+}
+
+func (pc *PlacementClient) updateRule(ctx context.Context, opType int) error {
+	url := pc.pdClient.GetLeaderAddr() + batchUpdatePrefix
+	header := http.Header{"Content-Type": {"application/json"}}
+	var content []byte
+	switch opType {
+	case updateTypeAdd:
+		content = []byte(addMetaJson)
+	case updateTypeDel:
+		content = []byte(delCDCGroupJson)
+	default:
+		return errors.Errorf("unexpected placement opration type: %d", opType)
+	}
+	_, err := pc.doRequest(ctx, url, http.MethodPost, header,
+		bytes.NewReader(content))
```

Comment:
```suggestion
	_, err := pc.doRequest(ctx, url, http.MethodPost, header, bytes.NewReader(content))
```

It's okay as long as it does not exceed 100.